contracts/optimistic-ethereum/OVM/execution/OVM_ExecutionManager.sol

```diff
@@ -1090,6 +1022,97 @@ contract OVM_ExecutionManager is iOVM_ExecutionManager, Lib_AddressResolver {
+            bool,
+            bytes memory
+        )
+    {
+        // Check that there is not already code at this address.
+        if (_hasEmptyAccount(_address) == false) {
+            return (
+                false,
+                _encodeRevertData(
+                    RevertFlag.CREATE_COLLISION,
+                    Lib_ErrorUtils.encodeRevertString("A contract has already been deployed to this address")
+                )
+            );
+        }
+
+        // Check the creation bytecode against the OVM_SafetyChecker.
+        if (ovmSafetyChecker.isBytecodeSafe(_creationCode) == false) {
+            return (
+                false,
+                _encodeRevertData(
+                    RevertFlag.UNSAFE_BYTECODE,
+                    Lib_ErrorUtils.encodeRevertString("Contract creation code contains unsafe opcodes.  Did you use the right compiler or pass an unsafe constructor argument?")
+                )
+            );
+        }
+
+        // We always need to initialize the contract with the default account values.
+        _initPendingAccount(_address);
+
+        // Actually execute the EVM create message,
+        address ethAddress = Lib_EthUtils.createContract(_creationCode);
+        
+        if (ethAddress == address(0)) {
+            // If the creation fails, the EVM lets us grab its revert data.  This may contain a revert flag
+            // to be used above in _handleExternalMessage.
+            uint256 revertDataSize;
+            assembly { revertDataSize := returndatasize() }
+            bytes memory revertdata = new bytes(revertDataSize);
+            assembly {
+                returndatacopy(
+                    add(revertdata, 0x20),
+                    0,
+                    revertDataSize
+                )
+            }
+            // Return that the creation failed, and the data it reverted with.
+            return (false, revertdata);
+        }
+
+        // Again simply checking that the deployed code is safe too. Contracts can generate
+        // arbitrary deployment code, so there's no easy way to analyze this beforehand.
+        bytes memory deployedCode = Lib_EthUtils.getCode(ethAddress);
+        if (ovmSafetyChecker.isBytecodeSafe(deployedCode) == false) {
+            return (
+                false,
+                _encodeRevertData(
+                    RevertFlag.UNSAFE_BYTECODE,
+                    Lib_ErrorUtils.encodeRevertString("Constrcutor attempted to deploy unsafe opcodes.")
+                )
+            );
+        }
+
+        // Contract creation didn't need to be reverted and the bytecode is safe. We finish up by
+        // associating the desired address with the newly created contract's code hash and address.
+        _commitPendingAccount(
+            _address,
+            ethAddress,
+            Lib_EthUtils.getCodeHash(ethAddress)
+        );
+
+        // Succressfuly deployments will not give acces to returndata, in both the EVM and the OVM.
```

Comment:
```suggestion
        // Successful deployments will not give access to returndata, in both the EVM and the OVM.
```

Reply:
Are we sure this is the case? Have we tested this? I don't think it particularly matters (not a consensus issue & I doubt anyone is actually writing contracts that use `RETURNDATACOPY` for successful contract creations.

Reply:
Yes, I have tested it a couple times.  This is why I didn't think the approach would work at all.

Reply:
I haven't looked for a reference regarding returndata from a `CREATE`, but my intuition about why you can't access it is because the RETURN opcode is used to get the actual runtime bytecode which is deployed. If one wanted _that_ info, they could use `EXTCODECOPY`.